File: benchmarks/src/ruby/task_scheduler.rb
T=300000;seed=12345;def ri(m);$seed=($seed*16807)%2147483647;$seed%m;end;$seed=12345
pq=[];qh=0;wq=[];wh=0;sq=[];sh=0;made=acc=sunk=0
start=Time.now.to_f*1000
T.times{made+=1
 if made%3!=0;wq<<made%100;end
 if wh<wq.length;pkt=wq[wh];wh+=1;acc=(acc+pkt*7)%1000003
  if pkt>50;sq<<pkt;end;end
 if sh<sq.length;sh+=1;sunk+=1;end}
finish=Time.now.to_f*1000
puts"Result: #{made}_#{acc}_#{sunk}";puts"Time: #{(finish-start).round}ms"

File: benchmarks/src/nodejs/task_scheduler.js
const T=300000;let seed=12345;function ri(m){seed=(seed*16807)%2147483647;return seed%m}
let pq=[],qh=0,wq=[],wh=0,sq=[],sh=0,made=0,acc=0,sunk=0;
const start=Date.now();
for(let t=0;t<T;t++){made++;if(made%3!=0)wq.push(made%100)
 if(wh<wq.length){let pkt=wq[wh];wh++;acc=(acc+pkt*7)%1000003;if(pkt>50)sq.push(pkt)}
 if(sh<sq.length){sh++;sunk++}}
const end=Date.now();console.log('Result: '+made+'_'+acc+'_'+sunk);console.log('Time: '+Math.floor(end-start)+'ms')
